controllers/neutronapi_controller.go

```diff
@@ -89,327 +98,467 @@ func (r *NeutronAPIReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
+		5,
+	)
 
-		err := controllerutil.SetControllerReference(instance, deployment, r.Scheme)
-		if err != nil {
-			return err
-		}
+	ctrlResult, err = depl.CreateOrPatch(ctx, helper)
+	if err != nil {
+		return ctrlResult, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+	// create Deployment - end
+
+	r.Log.Info("Reconciled Service successfully")
+	return ctrl.Result{}, nil
+}
+
+//
+// generateServiceConfigMaps - create create configmaps which hold scripts and service configuration
+// TODO add DefaultConfigOverwrite
+//
+func (r *NeutronAPIReconciler) generateServiceConfigMaps(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *neutronv1beta1.NeutronAPI,
+	envVars *map[string]common.EnvSetter,
+) error {
+	// Create/update configmaps from templates
+	cmLabels := common.GetLabels(instance, common.GetGroupLabel(neutronapi.ServiceName), map[string]string{})
+
+	// customData hold any customization for the service.
+	// custom.conf is going to /etc/<service>/<service>.conf.d
+	// all other files get placed into /etc/<service> to allow overwrite of e.g. logging.conf or policy.json
+	// TODO: make sure custom.conf can not be overwritten
+	customData := map[string]string{common.CustomServiceConfigFileName: instance.Spec.CustomServiceConfig}
+	for key, data := range instance.Spec.DefaultConfigOverwrite {
+		customData[key] = data
+	}
+
+	keystoneAPI, err := keystone.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
+	if err != nil {
+		return err
+	}
+	authURL, err := keystoneAPI.GetEndpoint(common.EndpointPublic)
+	if err != nil {
+		return err
+	}
+	templateParameters := make(map[string]interface{})
+	templateParameters["ServiceUser"] = instance.Spec.ServiceUser
+	templateParameters["KeystonePublicURL"] = authURL
 
+	ovnConnection, _, err := common.GetConfigMapAndHashWithName(ctx, h, instance.Spec.OVNConnectionConfigMap, instance.Namespace)
+	if err != nil {
+		return err
+	}
+	templateParameters["NBConnection"] = ovnConnection.Data["NBConnection"]
+	templateParameters["SBConnection"] = ovnConnection.Data["SBConnection"]
+
+	cms := []common.Template{
+		// ScriptsConfigMap
+		{
+			Name:               fmt.Sprintf("%s-scripts", instance.Name),
+			Namespace:          instance.Namespace,
+			Type:               common.TemplateTypeScripts,
+			InstanceType:       instance.Kind,
+			AdditionalTemplate: map[string]string{"init.sh": "/neutronapi/bin/init.sh"},
+			Labels:             cmLabels,
+		},
+		// ConfigMap
+		{
+			Name:          fmt.Sprintf("%s-config-data", instance.Name),
+			Namespace:     instance.Namespace,
+			Type:          common.TemplateTypeConfig,
+			InstanceType:  instance.Kind,
+			CustomData:    customData,
+			Labels:        cmLabels,
+			ConfigOptions: templateParameters,
+		},
+		// CustomConfigMap
+		{
+			Name:      fmt.Sprintf("%s-config-data-custom", instance.Name),
+			Namespace: instance.Namespace,
+			Type:      common.TemplateTypeCustom,
+			Labels:    cmLabels,
+		},
```

Comment:
I'm not sure you need this.  You're using the same pattern we have in the other modernized operators, where `custom.conf` is generated from the `CustomServiceConfig` CRD parameter.  In that case, the `custom.conf` is part of the `<instance.Name>-config-data` config map, so you don't need to mount an extra config map.  See https://github.com/openstack-k8s-operators/glance-operator for reference, as that uses the same pattern to handle custom config without requiring an extra config map for that custom config.

Reply:
And if you don't need it here, you can also remove all code elsewhere that references `<instance.Name>-config-data-custom`.

Reply:
Yes right, it's extra, cleaned all it's references.